voyager_donuts.py

```diff
@@ -1186,24 +1189,28 @@ def __receive_until_delim(self, delim=b'\r\n'):
             # seen issues with spamming thousands of [b'', b'' ...]
             if len(message_buffer) > 10 or len(self.message_overflow) > 10:
                 logging.fatal("Problem with receive_until_delim, exiting")
-                logging.fatal(f"meassge_buffer: {message_buffer}")
-                logging.fatal(f"meassge_overflow: {self.message_overflow}")
+                logging.fatal(f"message_buffer: {message_buffer}")
+                logging.fatal(f"message_overflow: {self.message_overflow}")
                 sys.exit(ERROR_UNHANDLED)
             # read the socket
             try:
-                message_raw = self.socket.recv(n_bytes)
+                message_raw = initial_message_raw + self.socket.recv(n_bytes)
             except s.timeout:
-                message_raw = b''
+                message_raw = initial_message_raw + b''
             logging.debug(f"Message raw {message_raw}")
 
+            if message_raw == b'':
+                continue
```

Comment:
To avoid spamming the code below with empty bytes, we can just skip early and go to the next iteration